### ISSUE TITLE
add node: users must explicitly set lat/lon

### ISSUE THREAD

**Issue by holgerd** [sozialhelden/wheelmap-classic#621]
As discussed in the [German OSM Forum](https://forum.openstreetmap.org/viewtopic.php?id=58514), some users have difficulty with positioning a new node on wheelmap at the correct lat/lon. 

**Behavior now:**
- I'm logged in on https://wheelmap.org/en/nodes/new/address?node=%5Bobject%20Object%5D :

<img width="953" alt="wheelmap_-_find_wheelchair_accessible_places" src="https://cloud.githubusercontent.com/assets/218506/26549369/02faf426-4479-11e7-8002-e0f474cddd71.png">

 - ..and type an address into the form. The map on the right hand side automatically updates its location and the marker is set to the center of that map:

<img width="950" alt="wheelmap_-_find_wheelchair_accessible_places" src="https://cloud.githubusercontent.com/assets/218506/26549401/2f69daf4-4479-11e7-86b4-100c3d272467.png">

**What I recommend changing:**

1. Add 2 titles to the columns: "1. address (optional)" and "2. point on the map (mandatory)"
2. Do not pre-set the marker. Set it only after the user has clicked on the map (and if s/he just clicks next without lat/lon just gets an error message <-- this already is ok)
3. change the help text to explain the new behavior.

<img width="950" alt="wheelmap_-_find_wheelchair_accessible_places" src="https://cloud.githubusercontent.com/assets/218506/26549773/e47fc66e-447a-11e7-8b12-07d117b8de61.png">

**Comment by lennerd** [comment]
Sound good to me. Maybe it make sense to first set the marker position before the user can type in an address?

**Comment by holgerd** [comment]
@lennerd Right now it's the other way round: entering the address will readjust the map so people can find the area easier. My hunch is that we should keep that.

**Comment by lennerd** [comment]
Ah I see. Than there is also a Geolocation of the user we can use to bring him near to the area he probably wants to add a marker to. But I see your point. Just feels strange to start with optional fields and have the mandatory second.

**Comment by 1000miles** [comment]
I do agree that `lat` and `lon` should be mandatory in order to find duplicates along with names faster. I also agree that having optional fields in the first place while having mandatory in the second place is somehow opposite to the usual user experience. 

**Some thoughts:**

1. How about changing the position of the map and address input form, so that `map` is left (please mark first) and `address` is right (edit the address)?

2. What is the value of keeping the address optional? 

If I am supposed to create a node I am usually aware of the `name`, `address` and `house number` (e.g. a coffee shop, etc.) or at least the street or area of the entrance (e.g. park) in order to be exact as I can, or? How can I mark a place if I am not aware of at least a specific area?

=> Isn't it better to request the user to be exact as they can from the beginning on and fix it later (because the place is identifiable even if the user uses just a random house number)?

**Comment by 1000miles** [comment]
Current status: Waiting for @holgerd to clarify the points/ideas we discussed yesterday.

**Comment by holgerd** [comment]
I suggest the following for the location selection in the "add node" flow:

# 1. new initial page
<img width="984" alt="balsamiq_mockups_for_desktop_-____users_hd_documents_sozialhelden_170608-add_node_bmml" src="https://user-images.githubusercontent.com/218506/26918541-c2e6e11c-4c32-11e7-8df3-34710afb6a1a.png">

- changed headline: "Where is this place? Click on the map to set the position"
- new: map on the left, address form on the right side
- new: omit the info text "You can move the marker to the correct location on the map."
- keep: map loads at users location if user gives permission in her browser
- keep: show error message if user clicks "next" without setting the marker on the map.
- omit additional info text (it's now the headline): "Bitte klicke in der Karte auf die Stelle, wo sich der Ort befindet!"
- keep: address fields are not mandatory

# 2. user clicked on the map

<img width="981" alt="balsamiq_mockups_for_desktop_-____users_hd_documents_sozialhelden_170608-add_node_bmml" src="https://user-images.githubusercontent.com/218506/26918143-08d0a480-4c31-11e7-91fe-44ee7c9a34f6.png">

- keep: address is auto-filled on click on the map. It stops to auto-fill as soon as the user changed one of the fields
- NEW: if user enters address in the form, it should have NO EFFECT to the position of the marker

**Comment by holgerd** [comment]
@1000miles @Svenyo @lennerd What do you think?

**Comment by Svenyo** [comment]
@holgerd 
Yes, I think the flow is good like that. 

Is there a reason why the headlines in the two pics differ (1. Click on the map to set its exact location, 2. Put a marker on the map)?

Suggestion for slight change in the headline: "Zoom in and click on the map to set its exact location"
If this is too long, maybe keep the info text that says instead: "Zoom in closer"

It could be useful to allow putting the marker only on a high zoom level where we can be sure the exact position can be set.

**Comment by lennerd** [comment]
I think the first flow looks good. I'm not sure about the second flow though as I would think it's dangerous because of the connection between map and pre-filling the address. Isn't that creating problems and inaccurate data again, because people rely on data on the map and do not check if the address is really correct?

**Comment by Svenyo** [comment]
@lennerd @holgerd I agree with Lennart that the automatic pre-filling could be dangerous. However it would be good to have an suggestion popping up in case the user made a spelling error. Then there could be a red line under the entered street name saying: "Did you mean .... ?"
Like that we could avoid both: wrong addresses and typing mistakes.

**Comment by lennerd** [comment]
> Then there could be a red line under the entered street name saying: "Did you mean .... ?"

👍

**Comment by holgerd** [comment]
@Svenyo Good catch. The headline was a mistake. It should be as written in the text: 

* changed headline: "Where is this place? Click on the map to set its exact location"


@lennerd and @Svenyo I saw that the OpenStreetMap editor iD is auto-guessing street names, but it happens in the form of auto-complete. So nothing is assumed beforehand. To de-risk and simplify this feature for now, I change my proposition regarding 2.:

- ~~keep:~~ **REMOVE:** address is auto-filled on click on the map. It stops to auto-fill as soon as the user changed one of the fields

means: never auto-fill an address.

**Comment by holgerd** [comment]
One addition: Map should default to zoom level 17 to make clear we want exact data.

**Comment by lennerd** [comment]
> I saw that the OpenStreetMap editor iD is auto-guessing street names, but it happens in the form of auto-complete.

Thats a very nice idea! :+1:

**Comment by 1000miles** [comment]
Following is a recap of the @TODO tasks for #621. 

@holgerd cc @Svenyo @anngreenberg 

Holger, could you please confirm the recap task list? 

# change

- [ ] user must click on marker first (mandatory), then on address field (optional) => change order: map to left, address fields to right
- [ ] change headline from `Wo liegt der Ort?` to `Wo liegt der Ort? Bitte klicke in die Karte, um den exakten Ort zu markieren.`
- [ ] remove additional text: `Du kannst den Marker auf der Karte auf die korrekte Position verschieben.`
- [ ] remove auto-fill completion in address fields
- [ ] do not allow to continue to next step (via `next` button) if no marker (step 1) has been set => show error message
- [ ] map should default to zoom level 17 

# keep

- [ ] if user approves `allow location access` the map loads the user's current geo location otherwise it will load the default location (currently Frankfurt am Main)
- [ ] address fields are optional

**Comment by holgerd** [comment]
Lgtm
